### PR TITLE
SDCICD-296: Refactor alert registration, add alerting script

### PR DESCRIPTION
**Pull request by jeefy**
This PR accomplishes 3 things:
- Add in alert script for Jenkins to call
- Fix Prometheus query logic to _actually work_
- Refactor alert registration (hindsight is 20/20)

### CODE REVIEW

File: pkg/common/alert/alert.go [hunk 1]
```diff
@@ -82,6 +82,7 @@ func (mas MetricAlerts) Notify() error {
 
 	promAPI := v1.NewAPI(client)
 	for _, ma := range mas {
+		log.Printf("Checking %s", ma.Name)
```
Comment (meowfaceman) [inline pkg/common/alert/alert.go:85]: Won't hold up the PR on this, but how easy would it be to move this to the metrics client and add the helper functions you need to support it?

File: scripts/granular-alerting.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+#
+# alert will look back at the previous job failures and generate granular alerts per-test-group
+#
+
+set -e
+
+#docker pull quay.io/app-sre/osde2e
```
Comment (meowfaceman) [inline scripts/granular-alerting.sh:8]: Uncomment this for the checkin.

File: pkg/common/alert/alert.go (hunk 1 above)
Comment (jeefy) [reply pkg/common/alert/alert.go:86]: Agree. :)

File: pkg/common/alert/alert.go [hunk 3]
```diff
@@ -75,14 +72,19 @@ type MetricAlert struct {
 
 // Notify prepares and then iterates through MetricAlerts to generate notifications
 func (mas MetricAlerts) Notify() error {
-	client, err := prometheus.CreateClient()
+	/*client, err := prometheus.CreateClient()
+	if err != nil {
+		return fmt.Errorf("unable to create Prometheus client: %v", err)
+	}*/
```
Comment (meowfaceman) [inline pkg/common/alert/alert.go:78]: nit: Remove. Otherwise lgtm.

Comment (jeefy) [reply pkg/common/alert/alert.go:78]: good catch

**Pull request merged**